rs-commerce/src/components/login-page-form/login-page-form.tsx

```diff
@@ -0,0 +1,187 @@
+  const [passwordTips, setPasswordTips] = useState('');
+  const [isOpenEye, setIsOpenEye] = useState(false);
+  const [isCanSubmit, setIsCanSubmit] = useState(false);
+
+  const checkIsEmailValid = (): void => {
+    if (emailTouched) {
+      if (checkIsValueEmpty(emailValue)) {
+        setEmailTips(EMAIL_FORM_TIPS.tipsEmailEmpty);
+        setIsEmailValid(false);
+      } else if (checkIsValueHasWhitespaces(emailValue)) {
+        setEmailTips(EMAIL_FORM_TIPS.tipsEmailWhitespaces);
+        setIsEmailValid(false);
+      } else if (!checkEmailHasAmpSymbol(emailValue)) {
+        setEmailTips(EMAIL_FORM_TIPS.tipsEmailAmpSymbol);
+        setIsEmailValid(false);
+      } else if (checkEmailHasLocalPart(emailValue)) {
+        setEmailTips(EMAIL_FORM_TIPS.tipsEmailLocalPart);
+        setIsEmailValid(false);
+      } else if (checkEmailLocalPartIsValid(emailValue)) {
+        setEmailTips(EMAIL_FORM_TIPS.tipsEmailLocalPartValid);
+        setIsEmailValid(false);
+      } else if (checkEmailHasDomain(emailValue)) {
+        setEmailTips(EMAIL_FORM_TIPS.tipsEmailDomain);
+        setIsEmailValid(false);
+      } else if (checkEmailDomainIsValid(emailValue)) {
+        setEmailTips(EMAIL_FORM_TIPS.tipsEmailDomainValid);
+        setIsEmailValid(false);
+      } else if (!checkEmailValid(emailValue)) {
+        setEmailTips(EMAIL_FORM_TIPS.tipsEmailFormat);
+        setIsEmailValid(false);
+      } else {
+        setEmailTips('');
+        setIsEmailValid(true);
+      }
+    }
+  };
+
+  const checkIsPasswordValid = (): void => {
+    if (passwordTouched) {
+      if (checkIsValueEmpty(passwordValue)) {
+        setPasswordTips(PASSWORD_FORM_TIPS.tipsPasswordEmpty);
+        setIsPasswordValid(false);
+      } else if (checkIsValueHasWhitespaces(passwordValue)) {
+        setPasswordTips(PASSWORD_FORM_TIPS.tipsPasswordWhitespaces);
+        setIsPasswordValid(false);
+      } else if (!checkIsValueHasLowercase(passwordValue)) {
+        setPasswordTips(PASSWORD_FORM_TIPS.tipsPasswordHasLowercase);
+        setIsPasswordValid(false);
+      } else if (!checkIsValueHasUppercase(passwordValue)) {
+        setPasswordTips(PASSWORD_FORM_TIPS.tipsPasswordHasUppercase);
+        setIsPasswordValid(false);
+      } else if (!checkIsValueHasDigit(passwordValue)) {
+        setPasswordTips(PASSWORD_FORM_TIPS.tipsPasswordHasDigit);
+        setIsPasswordValid(false);
+      } else if (!checkIsValueHasSpecial(passwordValue)) {
+        setPasswordTips(PASSWORD_FORM_TIPS.tipsPasswordHasSpecial);
+        setIsPasswordValid(false);
+      } else if (!checkIsValueHasOnlyLatin(passwordValue)) {
+        setPasswordTips(PASSWORD_FORM_TIPS.tipsPasswordHasOnlyLatin);
+        setIsPasswordValid(false);
+      } else if (checkPasswordLength(passwordValue)) {
+        setPasswordTips(PASSWORD_FORM_TIPS.tipsPasswordLength);
+        setIsPasswordValid(false);
+      } else {
+        setPasswordTips('');
+        setIsPasswordValid(true);
+      }
+    }
+  };
```

Comment:
the same like for checkIsEmailValid function

Reply:
I refactored it